WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4PeriodViewController.swift

```diff
@@ -763,34 +762,52 @@ private extension StoreStatsV4PeriodViewController {
+    func generateBarDataSet() -> LineChartData? {
         guard !orderStatsIntervals.isEmpty else {
             return nil
         }
 
         var barCount = 0
         var barColors: [UIColor] = []
-        var dataEntries: [BarChartDataEntry] = []
+        var dataEntries: [ChartDataEntry] = []
         let currencyCode = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
         orderStatsIntervals.forEach { (item) in
-            let entry = BarChartDataEntry(x: Double(barCount), y: (item.revenueValue as NSDecimalNumber).doubleValue)
+            let entry = ChartDataEntry(x: Double(barCount), y: (item.revenueValue as NSDecimalNumber).doubleValue)
             let currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
             let formattedAmount = currencyFormatter.formatHumanReadableAmount(String("\(item.revenueValue)"),
                                                                                 with: currencyCode,
                                                                                 roundSmallNumbers: false) ?? String()
             entry.accessibilityValue = "\(formattedChartMarkerPeriodString(for: item)): \(formattedAmount)"
-            barColors.append(.chartDataBar)
+            barColors.append(Constants.chartLineColor)
             dataEntries.append(entry)
             barCount += 1
         }
 
-        let dataSet = BarChartDataSet(entries: dataEntries, label: "Data")
-        dataSet.colors = barColors
-        dataSet.highlightEnabled = true
-        dataSet.highlightColor = .chartDataBarHighlighted
-        dataSet.highlightAlpha = Constants.chartHighlightAlpha
+        let hasRevenueData = hasRevenue()
+
+        let dataSet = LineChartDataSet(entries: dataEntries, label: "Data")
+        dataSet.drawCirclesEnabled = false
+        dataSet.colors = hasRevenueData ? barColors: .init(repeating: .clear, count: barColors.count)
+        dataSet.lineWidth = Constants.chartLineWidth
+        dataSet.highlightEnabled = hasRevenueData
+        dataSet.highlightColor = Constants.chartHighlightLineColor
+        dataSet.highlightLineWidth = Constants.chartHighlightLineWidth
         dataSet.drawValuesEnabled = false // Do not draw value labels on the top of the bars
-        return BarChartData(dataSet: dataSet)
+        dataSet.drawHorizontalHighlightIndicatorEnabled = false
+
+        // Configures gradient to fill the area from top to bottom when there is any positive revenue.
+        let hasNegativeRevenueOnly = orderStatsIntervals.map { $0.revenueValue }.contains(where: { $0 > 0 }) == false
+        if hasRevenueData && !hasNegativeRevenueOnly {
+            let gradientColors = [Constants.chartGradientBottomColor.cgColor, Constants.chartGradientTopColor.cgColor] as CFArray
+            let gradientColorSpace = CGColorSpaceCreateDeviceRGB()
+            let locations: [CGFloat] = hasNegativeRevenueOnly ? [1.0, 0.0]: [0.0, 1.0]
```

Comment:
`hasRevenueData` boolean indicates if there is any non-zero revenue data, so it could be all negative/positive or a mix of both 🙂 

https://github.com/woocommerce/woocommerce-ios/blob/2285a2bf8d3d1de81d9704e04453994bd7c0e826/WooCommerce/Classes/ViewRelated/Dashboard/Stats%20v4/StoreStatsV4PeriodViewController.swift#L758-L760

Reply:
Thanks for the explanation, but I think our check includes both `hasRevenue` and `!hasNegativeRevenueOnly`:

```Swift
if hasRevenueData && !hasNegativeRevenueOnly {
```

It's not a big issue though anyway, I'm just a bit confused here 😅

Reply:
sorry for the late reply! great catch, and I will update this in https://github.com/woocommerce/woocommerce-ios/issues/5745 🙂